src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -50,48 +54,161 @@ def total_avg_trade_price(markets):
     )
 
 
-def _store_bought_trade(result_dict, trade_offer):
-    result_dict['bought'] += trade_offer.energy
-    result_dict['spent'] += trade_offer.price / 100.
-    result_dict['total_energy'] += trade_offer.energy
-    result_dict['total_cost'] += trade_offer.price / 100.
+class MarketEnergyBills:
+    def __init__(self, is_spot_market=True):
+        self.is_spot_market = is_spot_market
+        self.bills_results = {}
+        self.bills_redis_results = {}
```

Comment:
Sorry for the nitpicking, but I'd prefer the name `bills_results_redis` to be consistent with the other variables in  enpoint_buffer.